src/lib.rs

```diff
@@ -43,16 +43,45 @@ fn cert_dirs_iter() -> impl Iterator<Item = &'static Path> {
         #[cfg(target_os = "haiku")]
         "/boot/system/data/ssl",
     ]
-    .iter().map(Path::new).filter(|p| p.exists())
+    .iter()
+    .map(Path::new)
+    .filter(|p| p.exists())
+}
+
+/// Deprecated as this isn't sound, use [`init_openssl_env_vars`] instead.
+#[doc(hidden)]
+#[deprecated(note = "this function is not safe, use `init_openssl_env_vars` instead")]
```

Comment:
Nitpicking about the message: it reads like "this function is unsafe, so please use the safe version, `init_openssl_env_vars`".
In reality what it means to say is "This function is not correctly marked as unsafe, use the correct version, `init_openssl_env_vars`, instead"